src/renderer/classes/Note.js

```diff
@@ -0,0 +1,61 @@
+import '../typedef';
+import { defaultNoteOn } from '../constants/defaults';
+import { pick } from 'lodash';
+
+const properties = ['time', 'pitch', 'power', 'timbre'];
+let id = 0;
+
+/**
+ * Instances of `Note` class are put in `store` as a state
+ * and will then be converted into MIDI
+ * */
+class Note {
+  /**
+   * @param {Modulation} noteOn
+   * */
+  constructor (noteOn) {
+    this.id = id;
+    id++;
+    /** @type {Array.<Modulation>} */
+    this.transition = [Object.assign({}, defaultNoteOn, pick(noteOn, properties))];
+  }
+  /**
+   * @param  {Array.<Modulation>} modulations
+   */
+  append (...modulations) {
+    modulations.forEach((modulation) => {
+      const modulationToAdd = Object.assign({}, pick(modulation, properties));
+      if (modulationToAdd.hasOwnProperty('time')) {
```

Comment:
Since `Object.prototype.hasOwnProperty` does not ensure that the value is not `null` or `undefined`, 
I would recommend replacing it with something like `if (modulationToAdd.time)` so that it checks that the key exists, and the value is not falsy.